Patch by MSP-Greg
Allows using mkmf.rb when running test-all tests from install.
mkmf.rb creates several global variables
diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index e4a6fc984f..033a78ec5e 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -228,6 +228,10 @@ def map_dir(dir, map = nil)
   elsif File.exist?(($hdrdir = ($top_srcdir ||= topdir) + "/include")  + "/ruby.h")
     $topdir ||= RbConfig::CONFIG["topdir"]
     $arch_hdrdir = "$(extout)/include/$(arch)"
+  elsif File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
+    $extmk = false
+    $topdir = $top_srcdir = $hdrdir
+    $arch_hdrdir = RbConfig::CONFIG["rubyarchhdrdir"]
   else
     abort <<MESSAGE
 mkmf.rb can't find header files for ruby at #{$hdrdir}/ruby.h
